velox/docs/functions/spark/decimal.rst

```diff
@@ -0,0 +1,8 @@
+=============================
+Decimal Functions
+=============================
+
+.. spark:function:: make_decimal(x, decimal, nullOnOverflow) -> decimal
+
+    Returns decimal value with the type of ``decimal`` from an unscaled bigint value ``x``,
```

Comment:
I'm not sure I understand this documentation. Would you add a link to Spark documentation and include an example?

Reply:
This is spark MakeDecimal(), https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/decimalExpressions.scala#L74
We interpret it as Spark function, actually, it is a Spark Expression, not spark function.

Reply:
We can see the example Spark test suite https://github.com/apache/spark/blob/master/sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/DecimalExpressionSuite.scala#L36

Reply:
Is this a special form similar to cast? If it cannot be implemented as cast, then we may need to extend Velox to allow for custom special forms.

Reply:
Yes, but it is different from `cast`. For example, cast input 101 will return Null, but MakeDecimal will return 10.1(in DecimalExpressionSuite)
```
spark-sql> select cast(101 as decimal(3,1));
NULL
Time taken: 0.982 seconds, Fetched 1 row(s)

spark-sql> select cast(10.1 as decimal(3,1));
10.1
Time taken: 0.921 seconds, Fetched 1 row(s)
```

Reply:
Sounds like this needs to be defined as a special form and we need to extend Velox to allow for custom special forms.

Reply:
Can we register it as spark function as temporary solution until custom special forms supported?

Reply:
And for gluten, we don't have substrait to custom special forms conversion. CC @FelixYBW

Reply:
Can we register make_decimal as `registerAllSpecialFormGeneralFunctions` in spark `register.cpp`? @mbasmanova 
```
// Special form functions don't have any prefix.
void registerAllSpecialFormGeneralFunctions() {
  VELOX_REGISTER_VECTOR_FUNCTION(udf_in, "in");
  VELOX_REGISTER_VECTOR_FUNCTION(udf_concat_row, "row_constructor");
  registerIsNullFunction("is_null");
}
```

Reply:
This function name is a bit misleading. IN, IS_NULL and row_constructor are special forms in Presto, but regular functions in Velox.

Reply:
So can I add a function `registerExpressionGeneralFunctions` for Spark Expression, make `make_decimal` a regular function in Velox?
And rename this function to `make_decimal_by_unscaled_value`.

Reply:
I took another look at this PR. It is quite hacky to pass a constant argument just to determine the output type of the function. If you really want to do that, I won't object. I do recommend to review what other expressions you want to add and come up with a cleaner design though.

Reply:
Yes, I agree. The more reasonable way is to put 2 integer argument named `precision` and `scale`, but velox not supports integer argument to generate return type in function signature.
This function also meets the same problem. https://github.com/facebookincubator/velox/issues/5598

Reply:
@jinchengchenghh, now that you have the special form for decimals, any thoughts on using that for this function?

Reply:
Sure, I will refactor this one.

Reply:
@majetideepak suggests the special form name to be `make_decimal` or `decimal_constructor`, not current `make_decimal_by_unscaled_value`, any thoughts on this? @mbasmanova

Reply:
@jinchengchenghh make_decimal sounds good. Let's also update PR title and description to clarify that make_decimal is a special form, not a function.

Reply:
@jinchengchenghh Doc was updated, but I'm not able to update PR title and description. Could you help when available?